bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Target.scala

```diff
@@ -24,32 +24,38 @@ object Target extends TargetLenses
 
 trait TargetLenses {
 
+  private def runTarget[A <: Target, B](l: A @> B):  A => IndexedStore[B, B, Target] =
+   (l.run _) andThen (_.map(x => x: Target))
+
+  private def pRunTarget[A <: Target, B](l: A @?> B): A => Option[Store[B, Target]] =
+   (l.run _) andThen (_.map(_.map(x => x: Target)))
+
   val name: Target @> String =
     Lens(_.fold(
-      TooTarget.name.run,
-      SiderealTarget.name.run,
-      NonSiderealTarget.name.run
+      runTarget(TooTarget.name),
+      runTarget(SiderealTarget.name),
+      runTarget(NonSiderealTarget.name)
     ))
```

Comment:
You had asked about this on iChat. If it compiles here we're fine, so let's leave it. Might make sense to switch to monocle lenses anyway.